mission_control/test/test_depth_heading_behavior.py

```diff
@@ -0,0 +1,166 @@
+    """ 
+        The test loads and execute a depth heading mission.
+        -   Load the mission depth_heading_mission_test.xml
+        -   Execute the mission
+        -   Check if the behavior publishes the goal
+        -   Simulate PoseEstimator data to finish the behavior
+        -   Test if the mission is SUCCESS
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        rospy.init_node('test_depth_heading_behavior')
+
+    def setUp(self):
+        mission_execute_state = ReportExecuteMissionState.PAUSED
+        mission_load_state = False
+        self.dir_path = os.path.dirname(os.path.abspath(__file__)) + '/test_files/'
+        mission_load_state_flag = False
+        report_mission_flag = False
+        report_mission = ReportMissions()
+        self.mission_load_state = None
+        self.report_mission = None
+        self.report_execute_mission = None
+        self.depth_heading_goal = []
+
+        # Subscribers
+        self.exec_state_sub = rospy.Subscriber('/mission_control_node/report_mission_execute_state',
+                                               ReportExecuteMissionState, self.callback_mission_execute_state)
+
+        self.load_state_sub = rospy.Subscriber('/mission_control_node/report_mission_load_state',
+                                               ReportLoadMissionState, self.callback_mission_load_state)
+
+        self.report_missions_sub = rospy.Subscriber('/mission_control_node/report_missions',
+                                                    ReportMissions, self.callback_report_mission)
+
+        self.depth_heading_msg = rospy.Subscriber('/mngr/depth_heading',
+                                                 DepthHeading, self.callback_publish_depth_heading_goal)
+
+        # Publishers
+        self.simulated_mission_control_load_mission_pub = rospy.Publisher('/mission_control_node/load_mission',
+                                                                          LoadMission, latch=True, queue_size=1)
+
+        self.simulated_mission_control_execute_mission_pub = rospy.Publisher(
+            '/mission_control_node/execute_mission', ExecuteMission, latch=True, queue_size=1)
+        
+        self.simulated_pose_estimator_pub = rospy.Publisher('/pose/corrected_data',
+                                                            CorrectedData, queue_size=1)
+
+    def callback_mission_execute_state(self, msg):
+        self.report_execute_mission = msg
+
+    def callback_report_mission(self, msg):
+        self.report_mission = msg
+
+    def callback_mission_load_state(self, msg):
+        self.mission_load_state = msg.load_state
+
+    def callback_publish_depth_heading_goal(self, msg):
+        self.depth_heading_goal = [msg.depth,
+                                  msg.heading,
+                                  msg.speed_knots,
+                                  msg.ena_mask]
+
+    def test_mission_with_depth_heading_behavior(self):
```

Comment:
@patapufate meta: same comment as in elsewhere about reusing code to reduce duplication.

Reply:
PTAL at #92. Then I'll implementet it here.